daml/Marketplace/Trading/Confirmation/Model.daml

```diff
@@ -47,6 +48,28 @@ template Confirmation
           accounts = (CustomerAccounts with customer = ctrl; ..) :: accounts
 
     controller provider can
+      ProcessCleared : ()
+        with
+          buyCid : ContractId Order.T
+          sellCid : ContractId Order.T
+        do
+          forA_ [buyCid, sellCid] $ \orderCid -> do
+            order <- fetchAndArchive orderCid
+            let
+              isPartialFill = execution.quantity < order.remainingQuantity
+              executions = execution :: order.executions
+              remainingQuantity = order.remainingQuantity - execution.quantity
+              quantities = [case order.details.side of
+                Order.Buy  -> execution.quantity * execution.price
+                Order.Sell -> execution.quantity]
+
+            if isPartialFill
+              then do
+                create order with executions, remainingQuantity, collateral = order.collateral, status = Order.PartiallyExecuted
```

Comment:
Do you need to set `collateral = order.collateral` since its already on the order?

```suggestion
                create order with executions, remainingQuantity, status = Order.PartiallyExecuted
```

Reply:
ah yes, good catch